### PR TITLE
[Draft] Limit length of System.Runtime.BigInteger to int.MaxValue-32 bits

### LINKED ISSUE #84670: IBinaryInteger<BigInteger>.GetShortestBitLength() - int overflow is not handled
### Description

When bit length of BigInteger value is greater than `2^(int.MaxValue+1) - 1`, then `IBinaryInteger<BigInteger>.GetShortestBitLength()` can not return correct result because bit length is more than int.MaxValue. 
Now `IBinaryInteger<BigInteger>.GetShortestBitLength()` returns incorrect negative value.

### Reproduction Steps

Code:
```csharp
using System.Numerics;
var l1 = 42;
var l2 = int.MaxValue;
BigInteger b = (BigInteger.One<<l1)<<l2;
var ll = 1L + l1 + l2;
Console.WriteLine(ll);
var res = ((IBinaryInteger<BigInteger>)b).GetShortestBitLength();
Console.WriteLine(res);
```

### Expected behavior

Exception on `GetShortestBitLength` call. Result can not be presented in `int` type.

### Actual behavior

output
```
2147483690
-2147483606
```

### Regression?

No

### Known Workarounds

No

### Configuration

The latest `main` branch

### Other information

probably from PR #69391
cc @tannergooding

It seems fix is quite trivial, I can do it.

### PR DESCRIPTION
**Pull request by speshuric**
See discussion in dotnet#84670

- Limit `BigInteger.MaxLength` to `int.MaxValue/32` uints. This is equal to limit length in bits to `int.MaxValue-32 bits`
- Fix ctor `BigInteger(ReadOnlySpan<byte>, bool, bool)` to check MaxLength limit
- Fix private ctor `BigInteger(ReadOnlySpan<uint>, bool)` to check `MaxLength` limit after trying to conserve space, not before
- Fix private ctor `BigInteger(Span<uint>)` to check MaxLength limit after trying to conserve space, not before
- Fix obsolete comment in AssertValid()
- Fix test `LargeNegativeBigIntegerShiftTest` to respect new limit.

fix dotnet#84670

### CODE REVIEW

**Comment by huoyaoyuan** [comment]
The overflow guard in `DebuggerDisplay` can also be updated.

**Comment by tannergooding** [comment]
Is this still being worked on? I'd ideally like to get this one merged prior to https://github.com/dotnet/runtime/pull/84733 or https://github.com/dotnet/runtime/pull/83951

File: src/libraries/System.Runtime.Numerics/src/System/Numerics/BigInteger.cs [hunk 1]
```diff
@@ -354,6 +354,10 @@ public BigInteger(ReadOnlySpan<byte> value, bool isUnsigned = false, bool isBigE
             {
                 int unalignedBytes = byteCount % 4;
                 int dwordCount = byteCount / 4 + (unalignedBytes == 0 ? 0 : 1);
+
+                if (dwordCount > MaxLength)
+                    ThrowHelper.ThrowOverflowException();
```
Comment (tannergooding) [inline src/libraries/System.Runtime.Numerics/src/System/Numerics/BigInteger.cs:359]: nit: Keep the braces please

**Comment by speshuric** [comment]
> Is this still being worked on? I'd ideally like to get this one merged prior to #84733 or #83951
Sorry, I have been busy a bit. I'll commit the rest of changes this week.

**Comment by adamsitnik** [comment]
> I'll commit the rest of changes this week.

Thank you for your help, there is obsoletely no rush from our side.

I am going to apply "need author action" label so this PR is not listed on our internal list of PRs awaiting for review.

**Comment by speshuric** [comment]
I still plan to complete this.

**Pull request closed without merging**